packages/manager/apps/dedicated/client/app/ip/firewall-game/rule/add/add.html

```diff
@@ -0,0 +1,99 @@
+                        for="protocol"
+                        class="control-label"
+                        data-translate="ip_game_mitigation_rule_add_protocol"
+                    >
+                    </label>
+                    <select
+                        required
+                        id="protocol"
+                        name="protocol"
+                        class="form-control"
+                        data-ng-model="$ctrl.rule.protocol"
+                        data-ng-options="getProtocoleText(protocol) for protocol in $ctrl.enums.protocols"
+                    >
+                    </select>
+                </div>
+
+                <div
+                    class="form-group"
+                    data-ng-class="getClassLabel(addRuleForm.portFrom)"
+                >
+                    <label
+                        for="portFrom"
+                        class="control-label"
+                        data-translate="ip_game_mitigation_rule_add_port_from"
+                    >
+                    </label>
+                    <input
+                        type="number"
+                        data-ng-model="$ctrl.rule.ports.from"
+                        class="form-control"
+                        required
+                        id="portFrom"
+                        name="portFrom"
+                        data-ng-min="1"
+                        data-ng-max="65535"
+                        data-ng-pattern="/^[0-9]*$/"
+                    />
+                    <div
+                        class="text-danger"
+                        data-ng-show="addRuleForm.portFrom.$invalid && addRuleForm.portFrom.$dirty && value == true && error !== 'required'"
+                        data-ng-repeat="(error, value) in addRuleForm.portFrom.$error"
+                        data-ng-bind="('ip_game_mitigation_rule_add_port_' + error) | ducTranslateAlt: error"
+                    ></div>
+                </div>
+
+                <div
+                    class="form-group"
+                    data-ng-class="getClassLabel(addRuleForm.portTo)"
+                >
+                    <label
+                        for="portTo"
+                        class="control-label"
+                        data-translate="ip_game_mitigation_rule_add_port_to"
+                    >
+                    </label>
+                    <input
+                        type="number"
+                        class="form-control"
+                        data-ng-model="$ctrl.rule.ports.to"
+                        id="portTo"
+                        name="portTo"
+                        data-ng-min="1"
+                        data-ng-max="65535"
+                        data-ng-pattern="/^[0-9]*$/"
+                    />
+                    <div
+                        class="error-label"
+                        data-ng-show="addRuleForm.portTo.$invalid && addRuleForm.portTo.$dirty && value == true"
```

Comment:
`value` can be different from boolean ?

Reply:
Don't look at the code, I've not rewritten everything :)